chainer/optimizer_hooks/gradient_hard_clipping.py

```diff
@@ -47,17 +46,4 @@ def __call__(self, rule, param):
         if grad is None:
             return
         with chainer.using_device(param.device):
-            xp = param.device.xp
-            if xp is backend.chainerx:
-                # TODO(kshitij12345): remove when chainerx.clip supports
-                # kwarg `out`.
-                param.grad = xp.clip(grad, self.lower_bound, self.upper_bound)
-            else:
-                if isinstance(grad, backend.intel64.mdarray):
-                    # Since np.clip doesn't support mdarray
-                    param.grad[param.grad <
-                               self.lower_bound] = self.lower_bound
-                    param.grad[param.grad >
-                               self.upper_bound] = self.upper_bound
-                    return
-                xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)
+            param.grad = param.grad.clip(self.lower_bound, self.upper_bound)
```

Comment:
If `grad` is not `intel64.mdarray`, `xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)` is more preferred because it can reduce memory consumption.

Reply:
Thanks for pointing. Fixed.